src/restconfig/src/main/java/org/geoserver/rest/catalog/GranuleHarvestingConfigurer.java

```diff
@@ -0,0 +1,76 @@
+/* (c) 2025 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.rest.catalog;
+
+import java.io.IOException;
+import java.util.Properties;
+import org.geoserver.rest.resources.GranuleHarvestingMetadata;
+import org.geotools.api.data.SimpleFeatureSource;
+import org.geotools.data.simple.SimpleFeatureCollection;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.GeometryFactory;
+
+/**
```

Comment:
Someone landing here first, or just wandering though the restconfig code, will not understand what this is for, the only harvesting done in restconfig is for image mosaics. I believe you need to clarify that it's for image mosaics, and justify why these classes are even here (I believe it's done to avoid dependency loops, so that dggs modules do not need to depend on vector-mosaic-rest, but it won't be evident to the random reader)